core_new/src/message/stored.rs

```diff
@@ -0,0 +1,172 @@
+// This file is part of Gear.
+
+// Copyright (C) 2022 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::identifiers::{MessageId, ProgramId};
+use crate::message::{
+    ContextStore, DispatchKind, ExitCode, GasLimit, IncomingDispatch, IncomingMessage, Payload,
+    Value,
+};
+use codec::{Decode, Encode};
+use core::ops::Deref;
+use scale_info::TypeInfo;
+
+/// Stored message.
+///
+/// Gasless Message for storing.
+#[derive(Clone, Default, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Decode, Encode, TypeInfo)]
+pub struct StoredMessage {
```

Comment:
Should we declare `StoredMessage` in the core? Or is it better to declare it on topper levels? Discussable question. As I see we will not use it in the core, isn't it?